lib/puppet/util/windows/file.rb

```diff
@@ -89,10 +89,31 @@ def symlink(target, symlink)
   end
   module_function :symlink
 
+
+  def file_exist(file_name)
+    path = file_name.to_s
```

Comment:
Because the call to `get_attributes` on line 97 expects a string.  To maintain compatibility with Ruby itself, `exist?` can be passed a `WatchedFile` instance, which needs `to_str` called against it to get the actual path string.

in other words, this line from the refactor never made it into this method:

```
path = path.to_str if path.respond_to?(:to_str) # support WatchedFile
```